lib/cloud_controller/diego/docker/protocol.rb

```diff
@@ -40,6 +40,8 @@ def desire_app_request(app, default_health_check_timeout)
         end
 
         def desire_app_message(app, default_health_check_timeout)
+          cached_docker_image = app.current_droplet.cached_docker_image if app.current_droplet
```

Comment:
I was able to comment out the `if app.current_droplet` from the code and no tests failed. Looks like you need a test there.